diff -up gazebo-4.0.2/gazebo/msgs/CMakeLists.txt.pic gazebo-4.0.2/gazebo/msgs/CMakeLists.txt
--- gazebo-4.0.2/gazebo/msgs/CMakeLists.txt.pic	2014-09-23 18:53:27.000000000 -0600
+++ gazebo-4.0.2/gazebo/msgs/CMakeLists.txt	2015-05-21 21:17:19.962116123 -0600
@@ -111,6 +111,7 @@ gz_build_tests(${msgs_tests_sources})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_executable(gazebomsgs_out generator/GazeboGenerator.cc generator/gazebo_generator.cc)
+set_target_properties(gazebomsgs_out PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_link_libraries(gazebomsgs_out ${PROTOBUF_LIBRARY} ${PROTOBUF_PROTOC_LIBRARY} pthread)
 
 set (sources msgs.cc MsgFactory.cc)
